### DISCUSSION TITLE
我的alist 启动 只能在CMD中输入命令server启动，用其他的启动方式都无法启动

### DISCUSSION THREAD

**Discussion by liucheng2046** [AlistGo/alist#3774] [Q&A]
我的alist 启动 只能在CMD中输入命令server启动，用其他的启动方式都无法启动。
已经尝试的方法有在和alist同一个文件夹用  .bat 启动，vbs启动，nssm添加服务启动都是无法使用。
用bat启动显示如下：
D:\alist>alist.exe server
INFO[2023-03-09 09:31:38] reading config file: data\config.json
INFO[2023-03-09 09:31:38] load config from env with prefix: ALIST_
INFO[2023-03-09 09:31:38] init logrus...
INFO[2023-03-09 09:31:38] start server @ 0.0.0.0:5244
INFO[2023-03-09 09:31:41] Shutdown Server ...
INFO[2023-03-09 09:31:42] timeout of 1 seconds.
INFO[2023-03-09 09:31:42] Server exiting

**如果是在cmd中用输入的命令行方式是可以成功启动的。**
cmd命令行启动显示的是：
C:\Users\liuch> D:\alist\alist.exe server
INFO[2023-03-09 09:33:43] reading config file: data\config.json
INFO[2023-03-09 09:33:43] load config from env with prefix: ALIST_
INFO[2023-03-09 09:33:43] init logrus...
INFO[2023-03-09 09:33:43] start server @ 0.0.0.0:5244
INFO[2023-03-09 09:33:43] success load storage: [/百度网盘], driver: [BaiduNetdisk]
INFO[2023-03-09 09:33:44] success load storage: [/阿里云盘], driver: [AliyundriveOpen]
INFO[2023-03-09 09:33:44] success load storage: [/一刻相册], driver: [BaiduPhoto]
INFO[2023-03-09 09:33:47] Shutdown Server ...
INFO[2023-03-09 09:33:47] Aria2 not ready.
INFO[2023-03-09 09:33:48] timeout of 1 seconds.
INFO[2023-03-09 09:33:48] Server exiting
只有这种方式能成功启动，每次输入特别麻烦

用VBS命令启动，启动后是无法登录账户的，同样的问题。
尝试用nssm加载服务的方式也是无法登录账户，同样的问题。

请大神帮忙解决，我只是想alist能随开机自动启动就行，已经试了几种方式都不行，只能用手动输入的。

**Comment by 2kil** [comment]
bat写:
start alist.exe server
试试

**Reply by liucheng2046** [reply]
非常感谢回复，刚测试了一下，用 start alist.exe server 还是不行：显示如下，还是加载不出来
INFO[2023-03-09 15:22:29] reading config file: data\config.json
INFO[2023-03-09 15:22:29] load config from env with prefix: ALIST_
INFO[2023-03-09 15:22:29] init logrus...
INFO[2023-03-09 15:22:29] start server @ 0.0.0.0:5244
INFO[2023-03-09 15:22:33] Aria2 not ready.
INFO[2023-03-09 15:22:34] qbittorrent not ready.

**Reply by 2kil** [reply]
是CMD还是powershell?

**Reply by liucheng2046** [reply]
都测试过了，问题都是一样的，只能用输入的方式才能正常启动。

**Reply by 2kil** [reply]
直接修改源代码吧，把启动参数去掉

**Reply by liucheng2046** [reply]
技术小白，修改源代码难度有点大😄。

**Reply by anwen-anyi** [reply]
windows10/11系统吗?

或者你试试文档里面提供的启动和停止脚本呢?

记得把脚本放到和AList软件本体同级目录喔

**Reply by andylawrence0079** [reply]
https://www.52pojie.cn/thread-1713978-1-1.html 到吾爱去看看,应该可以解决问题

**Comment by lianfan8937** [comment]
我跟你的问题一样，只要使用 BAT或者VBS脚本，alist目录下就会生成一个date文件夹，这时候你只要登录就会显示密码错误，搞得我现在PC上都无法自动启动，更不用说挂载到本地了。现在去吾爱论坛看看，研究一下，我在家是手机上挂载ALIST，电视上的KODI访问，就是感觉太手机耗电了。现在PC的感受还不如手机挂载alist呢。

**Reply by Rektyfikowany** [reply]
不加参数默认加载bat脚本当前目录下data文件夹，需要使用其它路径可以考虑`--data string`或者`--force-bin-dir`启动参数

**Reply by lianfan8937** [reply]
问题解决了  我没有在alist文件夹下启动cmd窗口

**Reply by liuyanhang2004** [reply]
cd to alist dir
./alist.exe start
